doc/protocolVersion_check.dox

```diff
@@ -0,0 +1,157 @@
+     Name     Major    Minor     Patch
+     [ver]      1        2         3
+\endverbatim
+
+An alternative method to communicate with a port in admin mode is to use telnet. <br>
+
+\verbatim
+    @terminal1$ yarp read /r
+    @terminal2$ telnet 127.0.0.1 10001    // where 127.0.0.1 : 10001 is the socket for the /r YARP port
+
+    Connected to 127.0.0.1
+    Escape character is '^]'.
+\endverbatim
+
+We can get access to YARP internal protocol by creating a connection to the admin port
+\verbatim
+    CONNECT myName      --> connect to the port by providing a custom user name
+    Welcome myName      --> response from the port
+    // list of available commands
+    ?
+    This is a YARP port.  Here are the commands it responds to:
+    *       Gives a description of this port
+    d       Signals the beginning of input for the port's owner
+    do      The same as "d" except replies should be suppressed ("data-only")
+    q       Disconnects
+    i       Interrupt parent process (unix only)
+    r       Reverse connection type to be a reader
+    /port   Requests to send output to /port
+    !/port  Requests to stop sending output to /port
+    ~/port  Requests to stop receiving input from /port
+    a       Signals the beginning of an administrative message
+    ?       Gives this help
+\endverbatim
+
+Writing the commad \c a we can have access to the admin functionalities through telnet connection.
+
+\verbatim
+    > a
+    > ver
+
+    >> Response
+   [ver] 1 2 4
+\endverbatim
+
+More details about the YARP connections and how they work can be found in the \ref yarp_protocol page.
+The protocol version number is composed by three values, called Major, Minor and Patch and it is stored in a structure called
+\ref ProtocolVersion: <br>
+- The Patch number is updated when a small fix or improvement is applied. <b> Patches must be backward compatible </b>.<br>
+- The Minor number is updated when a compatibility breaking change is applied to the protocol. <br>
+        Anyway the developer <b> must provide a backward compatibility layer for Minor changes</b>,
+        so that YARP communication between two machines running different versions of YARP protocol can still be succesfully performed.
+        See \ref deal_with_changes for more information. <br>
+- The Major number is updated when a new release of the protocol is delivered. Backward compatibility may or may not
+    be guaranteed, depending case by case. <br>
+
+
+\note Please note this is \b NOT the YARP library version, but it is a separated versioning specific for the port protocol. <br>
+YARP connections can be instantiate with different carriers, like tcp, udp, mjpeg, h264, portmonitor and many others.
+Users can write their own carriers and use them when connecting ports. The YARP protocol version is not related to any
+carrier used for the connection, instead is all about how YARP establish the communication and serialize the data, and it
```

Comment:
instead is all about how YARP establish the communication and serialize the data -> instead it is all about how YARP establishes the communication and serializes the data